### PR TITLE
Check all quotes for rfq before default quote query

### PR DESCRIPTION
**Pull request by abtestingalpha**
* Check all quotes for rfq, otherwise use best quote

## Summary by CodeRabbit

- **Refactor**
	- Enhanced quote management in the bridge interface for improved reliability.

3c6abd8cb784f34ed935c41c2282c53b98728784: [synapse-interface preview link ](https://sanguine-synapse-interface-obr2kmpvh-synapsecns.vercel.app)
60d802411e8c5eb39c824f8baf728114008275e5: [synapse-interface preview link ](https://sanguine-synapse-interface-dq7y3kw8w-synapsecns.vercel.app)
2b38d7901c1ee0436914a85899144854a27592ad: [synapse-interface preview link ](https://sanguine-synapse-interface-qvzopezdk-synapsecns.vercel.app)
242fbcb946fb4613d2c1d82eda161f07d5d19bc9: [synapse-interface preview link ](https://sanguine-synapse-interface-i7l11i962-synapsecns.vercel.app)

### CODE REVIEW

File: packages/synapse-interface/pages/state-managed-bridge/index.tsx [hunk 1]
```diff
@@ -174,6 +166,20 @@ const StateManagedBridge = () => {
         (quote) => quote.bridgeModuleName === 'SynapseRFQ'
       )
 
+      let quote
+
+      if (rfqQuote) {
+        quote = rfqQuote
+      } else {
+        quote = await synapseSDK.bridgeQuote(
```
Comment (ChiTimesChi) [inline packages/synapse-interface/pages/state-managed-bridge/index.tsx:174]: You can go even further and strip this call altogether.

allBridgeQuotes returns the full list of quotes sorted by destination amount out, so picking the first entry from the list should effectively do the same thing as issuing another bridgeQuote

Comment (ChiTimesChi) [reply packages/synapse-interface/pages/state-managed-bridge/index.tsx:174]: This reminds me that SDK documentation hasn't been updated for a loooong time - aiming to do this next week

**Review by ChiTimesChi — CHANGES_REQUESTED** [review]
Requesting a change to optimize even further

File: packages/synapse-interface/pages/state-managed-bridge/index.tsx (hunk 1 above)
Comment (abtestingalpha) [reply packages/synapse-interface/pages/state-managed-bridge/index.tsx]: Updated [here](https://github.com/synapsecns/sanguine/pull/1861/commits/0adc35cb771bd4d59ed61e0442b3e883503c09a6)

File: packages/synapse-interface/pages/state-managed-bridge/index.tsx [hunk 2]
```diff
@@ -174,6 +166,15 @@ const StateManagedBridge = () => {
         (quote) => quote.bridgeModuleName === 'SynapseRFQ'
       )
 
+      let quote
```
Comment (ChiTimesChi) [inline packages/synapse-interface/pages/state-managed-bridge/index.tsx:169]: Do we want to check `allQuotes` for not being an empty list here?

`allBridgeQuotes` doesn't throw an error, if no quotes have been found and just returns an empty list instead.

Comment (ChiTimesChi) [reply packages/synapse-interface/pages/state-managed-bridge/index.tsx:169]: This being said, the test build still handles the empty list case correctly, as the error is thrown here (`quote` is undefined there)

https://github.com/synapsecns/sanguine/blob/0adc35cb771bd4d59ed61e0442b3e883503c09a6/packages/synapse-interface/pages/state-managed-bridge/index.tsx#L178-L186

Comment (abtestingalpha) [reply packages/synapse-interface/pages/state-managed-bridge/index.tsx:174]: Yep, makes sense, handling [here](https://github.com/synapsecns/sanguine/pull/1861/commits/4e7c89a752645705149bb48907ce057420170ab9)

Comment (ChiTimesChi) [reply packages/synapse-interface/pages/state-managed-bridge/index.tsx:174]: Perfect, thanks!

**Pull request merged**